hibernate-core/src/main/java/org/hibernate/metamodel/mapping/internal/ToOneAttributeMapping.java

```diff
@@ -998,6 +998,22 @@ public EntityFetch generateFetch(
 				|| fetchParent.getNavigablePath() instanceof TreatedNavigablePath
 				&& parentNavigablePath.equals( fetchParent.getNavigablePath().getRealParent() );
 
+		if ( cardinality == Cardinality.ONE_TO_ONE && bidirectionalAttributeName != null && fetchParent.getReferencedMappingType() == getEntityMappingType() ) {
+			return new EntityFetchSelectImpl(
+					fetchParent,
+					this,
+					fetchablePath,
+					foreignKeyDescriptor.createDomainResult(
+							fetchablePath,
+							parentTableGroup,
+							ForeignKeyDescriptor.Nature.TARGET,
```

Comment:
It seems to me, that this is not entirely correct. See how the side nature is determined below.

Reply:
if I use the same way, the test fails may be the issue is the wrong value of the `this.sideNature`